src/main/webapp/app/quiz/participate/quiz.controller.js

```diff
@@ -0,0 +1,131 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('artemisApp')
+        .controller('QuizController', QuizController);
+
+    QuizController.$inject = ['$scope', '$stateParams', '$interval', 'QuizExerciseForStudent', 'QuizSubmission', 'QuizSubmissionForExercise'];
+
+    function QuizController($scope, $stateParams, $interval, QuizExerciseForStudent, QuizSubmission, QuizSubmissionForExercise) {
+        var vm = this;
+
+        var timeDifference = 0;
+
+        vm.isSubmitting = false;
+        vm.lastSubmissionTimeText = "never";
+
+        vm.remainingTime = "?";
+        vm.remainingTimeSeconds = 0;
+
+        // update displayed times in UI regularly
+        $interval(function () {
+            // update remaining time
+            if (vm.quizExercise && vm.quizExercise.adjustedDueDate) {
+                var endDate = vm.quizExercise.adjustedDueDate;
+                if (endDate.isAfter(moment())) {
+                    vm.remainingTimeSeconds = endDate.diff(moment(), "seconds");
+                    if (vm.remainingTimeSeconds > 90) {
+                        vm.remainingTime = Math.ceil(vm.remainingTimeSeconds / 60) + " minutes";
```

Comment:
Why do you not show the seconds earlier? You could at least show e.g. "1 min 30 sec" if it is below 3 minutes. I would also abbreviate minutes to "min" and seconds to "sec" or even "s"